File: library/src/main/java/com/juanimoli/strictmodenotifier/detector/ResourceMismatchDetector.java
package com.juanimoli.strictmodenotifier.detector;

import com.juanimoli.strictmodenotifier.StrictModeLog;

public class ResourceMismatchDetector implements Detector {

    @Override
    public boolean detect(StrictModeLog log) {
        return log.message.contains("StrictMode$StrictModeResourceMismatchViolation");
    }
}

File: notifier/src/main/java/io/github/juanimoli/strictmode/notifier/detector/ResourceMismatchDetector.kt
package io.github.juanimoli.strictmode.notifier.detector

import io.github.juanimoli.strictmode.notifier.StrictModeLog

class ResourceMismatchDetector : Detector {
    private val detectMessageList = listOf(
        "StrictMode(.*)StrictModeResourceMismatchViolation".toRegex(),
        "StrictMode(.*)ResourceMismatchViolation".toRegex()
    )

    override fun detect(log: StrictModeLog): Boolean {
        return detectMessageList.any { it.find(log.message) != null }
    }
}
